qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -159,11 +167,37 @@ def run(self, dag):
                         dag.remove_op_node(node)
                 else:
                     dag.replace_block_with_op(run, unitary, {qubit: 0}, cycle_check=False)
+
+        dag = self._handle_control_flow_ops(dag)
+
         # Clear collected blocks and runs as they are no longer valid after consolidation
         if "run_list" in self.property_set:
             del self.property_set["run_list"]
         if "block_list" in self.property_set:
             del self.property_set["block_list"]
+
+        return dag
+
+    def _handle_control_flow_ops(self, dag):
+        """
+        This is similar to transpiler/passes/utils/control_flow.py except that the
+        collect blocks is redone for the control flow blocks.
+        """
+        from qiskit.transpiler import PassManager
```

Comment:
If you make the import `from qiskit.transpiler.passmanager import PassManager` you can probably avoid pylint complaining about the cyclic import, and be able to move this up to the top level imports.

Reply:
Done in dc69654